### PR TITLE
Oh hai template snippets

### PR DESCRIPTION
**Pull request by sglanzer-fire**
# fix
# CHANGELOG

Added ember-code-snippets to allow demo documentation to reference the code as the sample text.  Cleaned up the button demo as an example.

### CODE REVIEW

File: tests/dummy/app/pods/button/template.hbs [hunk 1]
```diff
@@ -7,533 +7,416 @@
...
       </div>
       <div class='snippet'>
-        {{format-markdown "```handlebars
-{{frost-button
-  priority='primary'
-  size='small'
-  text='Text'
-}}
-```"}}
+        {{code-snippet name='button-primary-small.hbs'}}
```
Comment (job13er) [inline tests/dummy/app/pods/button/template.hbs:25]: OK, this is kinda weird, kinda cool. There is no `button-primary-small.hbs` file! 😱

Comment (sglanzer-fire) [inline tests/dummy/app/pods/button/template.hbs:25]: Witchcraft!

Comment (sglanzer-fire) [inline tests/dummy/app/pods/button/template.hbs:25]: They use the extension to tell highlight.js how to deal with the code and there are two styles of snippet supported - inline and file based.  So they just load up all of the files and inline flags in memory - if the inline snippet occurs in an hbs file then it gets an hbs extension in the mapping

**Comment by sglanzer-fire** [comment]
I'm going to ask @EWhite613 to make the highlighter a bit prettier when he frees up - should be able to match most of what we have in the atom/vsc/brackets highlighters

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/139/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**